### PR TITLE
Add docs for feature reset API

### PR DESCRIPTION
**Pull request by williamrandolph**
Here are some docs for the Feature Reset API. These should have been included in #69469, but were left for later.

### CODE REVIEW

**Comment by williamrandolph** [comment]
I have two questions about this work:
- [ ] How can I create a "warning" or "important" block to emphasize that this API shouldn't be used on production systems?
- [ ] Is there a way to loosen the check on the API response so that this test doesn't break when feature states added or removed.

**Comment by jaymode** [comment]
> How can I create a "warning" or "important" block to emphasize that this API shouldn't be used on production systems?

For the warning, I am guessing something like the one on this page is good: https://github.com/elastic/elasticsearch/blob/master/docs/reference/index-modules.asciidoc

**Comment by jaymode** [comment]
> Is there a way to loosen the check on the API response so that this test doesn't break when feature states added or removed.

Maybe the following from https://github.com/elastic/elasticsearch/tree/master/docs#readme will be helpful:

> One interesting difference here is that you often want to match against the response from Elasticsearch. To do that you can reference the "body" of the response like this: // TESTRESPONSE[s/"took": 25/"took": $body.took/]

**Comment by williamrandolph** [comment]
I played with the substitutions, but I couldn't find a way to indicate a variable number of entries in a list. So we can either skip checking the result of this test, or we can require that if we add a feature state to the system, we have to add an entry here. Thoughts?

File: docs/reference/features/apis/reset-features-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+[[reset-features-api-desc]]
+==== {api-description-title}
+
+A feature state includes one or more system indices necessary for a given
+feature to function. Built-in features and features defined by plugins may
+or may not provide endpoints for resetting the state of those features, so
+this endpoint provides a high-level way of clearing cluster state.
+
+Use the <<get-features-api,get features API>> to see a list of features that
+will be affected by this endpoint. This API clears all features.
```
Comment (debadair) [inline docs/reference/features/apis/reset-features-api.asciidoc:35]:
```suggestion
Return a cluster to the same state as a new installation by resetting the feature state for all {es} features. This deletes all state information stored in system indices.

Note that select features might provide a way to reset particular system indices. Using this API resets _all_ features, both those that are built-in and implemented as plugins. 

To list the features that will be affected, use the <<get-features-api,get features API>>. 
```

File: docs/reference/features/apis/reset-features-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+WARNING: The list of features that will be cleared comes from the node against
+which you issue this command, so if you have any doubts about which nodes have
+certain plugins installed, you will want to issue your command against the
+master node.
+
+==== {api-examples-title}
+
```
Comment (debadair) [inline docs/reference/features/apis/reset-features-api.asciidoc:43]:
```suggestion
Example response:
```

**Review by debadair — COMMENTED** [review]
Left some suggestions--let me know if you have any questions!

**Comment by jaymode** [comment]
I think you should be able to do what we do for things like plugins:

```
// TESTRESPONSE[s/"plugins": \[[^\]]*\]/"plugins": $body.$_path/]
```

**Review by jaymode — APPROVED** [review]
Docs look good aside for Deb's suggestions.

**Comment by williamrandolph** [comment]
Backport commit: a46b8eaea958b5cad61cf30993d4d12fe9b0b13b

**Pull request merged**